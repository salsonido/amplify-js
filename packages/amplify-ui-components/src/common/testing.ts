import {
	boolean as booleanKnob,
	select as selectKnob,
	text as textKnob,
} from '@storybook/addon-knobs';

export const pixelThreshold = 10;

export const selectors = {
	// amplify-form-field
	formFieldDescription: '[data-test=form-field-description]',
};

export const knobs = {
	labelKnob(labelPlaceholder: string) {
		return textKnob('Label', labelPlaceholder);
	},

	descriptionKnob(descriptionPlaceholder: string) {
		return textKnob('Description', descriptionPlaceholder);
	},

	hintKnob(hintPlaceholder: string) {
		return textKnob('Hint', hintPlaceholder);
	},

	inputTypeKnob(inputTypePlaceholder: string) {
		return selectKnob(
			'Type',
			[
				'date',
				'email',
				'number',
				'password',
				'search',
				'tel',
				'text',
				'url',
				'time',
			],
			inputTypePlaceholder
		);
	},

	placeholderKnob(placeholderPlaceholder: string) {
		return textKnob('Placeholder', placeholderPlaceholder);
	},

	buttonTypeKnob(buttonTypePlaceholder: string) {
		return selectKnob(
			'Button type',
			['button', 'submit', 'reset'],
			buttonTypePlaceholder
		);
	},

	buttonTextKnob(buttonTextPlaceholder: string) {
		return textKnob('Button text', buttonTextPlaceholder);
	},

	toggleKnob(toggleDefault: boolean) {
		return booleanKnob('Toggle', toggleDefault);
	},

	disabledKnob(disabledDefault: boolean) {
		return booleanKnob('Disabled', disabledDefault);
	},

	formSectionHeaderKnob(headerPlaceholder: string) {
		return textKnob('Header text', headerPlaceholder);
	},

	formSectionSubmitKnob(submitPlaceholder: string) {
		return textKnob('Submit button text', submitPlaceholder);
	},

	tooltipAutoShowKnob(autoShowDefault: boolean) {
		return booleanKnob('Tooltip autoshow', autoShowDefault);
	},
};
