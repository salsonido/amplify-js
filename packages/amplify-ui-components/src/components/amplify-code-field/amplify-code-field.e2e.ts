import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-code-field e2e:', () => {
	it('should render a amplify-code-field', async () => {
		const page = await newE2EPage();
		await page.setContent(`<amplify-code-field></amplify-code-field>`);
		const el = await page.find('amplify-code-field');
		expect(el).not.toBeNull();

		const screenshot = await page.compareScreenshot('amplify-code-field', {
			fullPage: true,
		});
		expect(screenshot).toMatchScreenshot({
			allowableMismatchedPixels: pixelThreshold,
		});
	});
});
