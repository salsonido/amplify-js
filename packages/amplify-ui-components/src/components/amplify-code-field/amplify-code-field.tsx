import { I18n } from '@aws-amplify/core';
import { Component, Prop, FunctionalComponent, h } from '@stencil/core';
import { CODE_SUFFIX } from '../../common/constants';
import { Translations } from '../../common/Translations';

@Component({
	tag: 'amplify-code-field',
})
export class AmplifyCodeField {
	/** Based on the type of field e.g. sign in, sign up, forgot password, etc. */
	@Prop() fieldId: string = CODE_SUFFIX;
	/** Used for the code label */
	@Prop() label: string = Translations.CODE_LABEL;
	/** Used for the placeholder label */
	@Prop() placeholder: string = Translations.CODE_PLACEHOLDER;
	/** Used as the hint in case you forgot your confirmation code, etc. */
	@Prop() hint: string | FunctionalComponent | null;
	/** The required flag in order to make an input required prior to submitting a form */
	@Prop() required: boolean = false;
	/** The callback, called when the input is modified by the user. */
	@Prop() handleInputChange?: (inputEvent: Event) => void;
	/** The value of the content inside of the input field */
	@Prop() value?: string;
	/** Attributes places on the input element: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes */
	@Prop() inputProps?: object;
	/** Will disable the input if set to true */
	@Prop() disabled?: boolean;

	render() {
		return (
			<amplify-form-field
				fieldId={this.fieldId}
				label={I18n.get(this.label)}
				placeholder={I18n.get(this.placeholder)}
				type="number"
				hint={this.hint}
				required={this.required}
				handleInputChange={this.handleInputChange}
				value={this.value}
				inputProps={this.inputProps}
				disabled={this.disabled}
			/>
		);
	}
}
