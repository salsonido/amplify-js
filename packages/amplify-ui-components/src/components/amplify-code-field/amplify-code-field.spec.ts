import { newSpecPage } from '@stencil/core/testing';
import { AmplifyCodeField } from './amplify-code-field';

describe('amplify-code-field spec:', () => {
	describe('Component logic ->', () => {
		let codeField;

		beforeEach(() => {
			codeField = new AmplifyCodeField();
		});

		it('should render `code` for the field id', () => {
			expect(codeField.fieldId).toEqual('code');
		});

		it('should have `label` render to `Verification code` by default', () => {
			expect(codeField.label).toEqual('Verification code');
		});

		it('should have `placeholder` render to `Enter code` by default', () => {
			expect(codeField.placeholder).toEqual('Enter code');
		});

		it('should have `required` set to `false` by default', () => {
			expect(codeField.required).toBe(false);
		});
	});
	describe('Render logic ->', () => {
		it('should render `placeholder` and `label values', async () => {
			const page = await newSpecPage({
				components: [AmplifyCodeField],
				html: `<amplify-code-field label="New code *" placeholder="Enter new code"></amplify-code-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});

		it('should render default values by default', async () => {
			const page = await newSpecPage({
				components: [AmplifyCodeField],
				html: `<amplify-code-field></amplify-code-field>`,
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});
