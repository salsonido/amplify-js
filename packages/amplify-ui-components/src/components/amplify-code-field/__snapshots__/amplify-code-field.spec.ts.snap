// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-code-field spec: Render logic -> should render \`placeholder\` and \`label values 1`] = `
<amplify-code-field label="New code *" placeholder="Enter new code">
  <amplify-form-field fieldid="code" label="New code *" placeholder="Enter new code" type="number"></amplify-form-field>
</amplify-code-field>
`;

exports[`amplify-code-field spec: Render logic -> should render default values by default 1`] = `
<amplify-code-field>
  <amplify-form-field fieldid="code" label="Verification code" placeholder="Enter code" type="number"></amplify-form-field>
</amplify-code-field>
`;
